### ISSUE TITLE
Spread operator should be allowed anywhere in array literals and function calls

### ISSUE THREAD

**Issue by BenoitZugmeyer** [eslint/espree#89]
Those statements should be valid:
```js
var a = [...b, 1];
foo(...a, b, ...c);
```
Right now, running eslint on code like this gives me `error  Spread must be the final element of an element list` and as it is a syntax error, it does not lint the file.

See [MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_operator)

**Comment by nzakas** [comment]
Oops, must be confusing named function argument restrictions with other uses.

**Comment by nzakas** [comment]
Working on this.

**Comment by BenoitZugmeyer** [comment]
Awesome, thank you !
